src/ledger/submit.ts

```diff
@@ -0,0 +1,64 @@
+import { encode } from 'ripple-binary-codec'
+
+import type { Client, SubmitRequest, SubmitResponse, Wallet } from '..'
+import { RippledError } from '../common/errors'
+import { Transaction } from '../models/transactions'
+import { sign } from '../wallet/signer'
+
+import autofill from './autofill'
+
+/**
+ * Submits an unsigned transaction.
+ * Steps performed on a transaction:
+ *    1. Autofill.
+ *    2. Sign & Encode.
+ *    3. Submit.
+ *
+ * @param client - A Client.
+ * @param wallet - A Wallet to sign a transaction.
+ * @param transaction - A transaction to autofill, sign & encode, and submit.
+ * @returns A promise that contains SubmitResponse.
+ * @throws RippledError if submit request fails.
+ */
+async function submitTransaction(
+  client: Client,
+  wallet: Wallet,
+  transaction: Transaction,
+): Promise<SubmitResponse> {
+  const tx: Transaction = await autofill(client, transaction)
+  const signedTxSerialized: string = sign(wallet, tx)
+  return submitRequest(client, signedTxSerialized)
+}
+
+/**
+ * Encodes and submits a signed transaction.
+ *
+ * @param client - A Client.
+ * @param signedTransaction - A signed transaction to encode and submit.
+ * @returns A promise that contains SubmitResponse.
+ * @throws RippledError if submit request fails.
+ */
+async function submitSignedTransaction(
+  client: Client,
+  signedTransaction: Transaction,
```

Comment:
We could have this take either a `string` or a `Transaction`, so the encoding is optional and only done if it's a `Transaction`